FlaxGPTNeo Draft PR

zanussbaum opened the pull request:
# What does this PR do?
Add FlaxGPTNeo to HuggingFace Models!
<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- albert, bert, xlm: @LysandreJik
- blenderbot, bart, marian, pegasus, encoderdecoder,  t5: @patrickvonplaten, @patil-suraj
- longformer, reformer, transfoxl, xlnet: @patrickvonplaten
- fsmt: @stas00
- funnel: @sgugger
- gpt2: @patrickvonplaten, @LysandreJik
- rag: @patrickvonplaten, @lhoestq
- tensorflow: @LysandreJik

Library:

- benchmarks: @patrickvonplaten
- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam
- text generation: @patrickvonplaten
- tokenizers: @n1t0, @LysandreJik
- trainer: @sgugger
- pipelines: @LysandreJik

Documentation: @sgugger

HF projects:

- datasets: [different repo](https://github.com/huggingface/datasets)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj
- research_projects/bert-loses-patience: @JetRunner
- research_projects/distillation: @VictorSanh

 -->

zanussbaum committed:
blank file for draft pr

patrickvonplaten commented:
Hey @zanussbaum, 

Regarding on how to proceed with the implementation, could you maybe post your questions here and tag @patil-suraj and @patrickvonplaten so that we can move forward? :-)

zanussbaum commented:
Hey @patrickvonplaten, I actually chatted with Suraj this morning and
cleared my questions up about the Self Attention Module. I am working on
implementing it and hope to have something out this weekend!

On Thu, Jun 3, 2021 at 11:27 AM Patrick von Platen ***@***.***>
wrote:

> Hey @zanussbaum <https://github.com/zanussbaum>,
>
> Regarding on how to proceed with the implementation, could you maybe post
> your questions here and tag @patil-suraj <https://github.com/patil-suraj>
> and @patrickvonplaten <https://github.com/patrickvonplaten> so that we
> can move forward? :-)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/huggingface/transformers/pull/11909#issuecomment-853957259>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBFIPPVDY3FYPLTVEYQCTTTQ6NPXANCNFSM45U7XKKQ>
> .
>

zanussbaum committed:
[WIP] Boilerplate Flax GPT Neo

This adds the FlaxGPTNeo implementation (still WIP). Adding for reference
and debugging purposes

zanussbaum committed:
Adding exports
